core/include/gnuradio-4.0/Transactions.hpp

```diff
@@ -42,20 +42,46 @@ class CtxSettings : public SettingsBase {
      * The predicate will be called until it returns "true" (a match is found), or until it returns std::nullopt,
      * which indicates that no matches were found and there is no chance of matching anything in a further round.
      */
-    using MatchPredicate = std::function<std::optional<bool>(const property_map &, const property_map &, std::size_t)>;
-
-    TBlock                                       *_block = nullptr;
-    mutable std::mutex                            _lock{};
-    property_map                                  _active{};
-    property_map                                  _staged{};
-    std::set<std::string, std::less<>>            _auto_update{};
-    std::set<std::string, std::less<>>            _auto_forward{};
-    std::unordered_map<SettingsCtx, property_map> _settings{};
-    property_map                                  _default_settings{};
-    MatchPredicate                                _match_pred = nullMatchPred;
+    using MatchPredicate = std::function<std::optional<bool>(const pmtv::pmt &, const pmtv::pmt &, std::size_t)>;
```

Comment:
N.B. the follow-up PR should replace this with the new matcher structure. See [here](https://github.com/fair-acc/gnuradio4/blob/filterStreamToDataSet/core/include/gnuradio-4.0/TriggerMatcher.hpp) for details.

Reply:
Added to the list of follow-up issues.